### PR TITLE
Sg2/graph manip api work

### PR DESCRIPTION
**Pull request by elizabeth-legros**

### CODE REVIEW

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs [hunk 1]
```diff
@@ -146,40 +146,27 @@ public ShaderFunction GetShaderFunction(INodeReader data, ShaderContainer contai
...
-                portWriter = nodeWriter.AddPort<GraphType>(userData, "Exp", true, registry);
-                portWriter.SetField<int>(GraphType.kLength, 1);
-                portWriter = nodeWriter.AddPort<GraphType>(userData, "Out", false, registry);
-                portWriter.SetField<int>(GraphType.kLength, 1);
+                var port = node.AddPort("In", true, true);
+                port.AddField(GraphType.kLength, 1);
+                port = node.AddPort("Exp", true, true);
+                port.AddField(GraphType.kLength, 1);
+                port = node.AddPort("Out", false, true);
+                port.AddField(GraphType.kLength, 1);
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs]: Does this do the same GraphType setup that AddPort<T> did before?

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/PortHandler.cs [hunk 2]
```diff
@@ -1,9 +1,19 @@
 using System.Collections.Generic;
+using UnityEditor.ContextLayeredDataStorage;
 
 namespace UnityEditor.ShaderGraph.GraphDelta
 {
     public class PortHandler : GraphDataHandler
     {
+        public string LocalID { get; private set; }
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/PortHandler.cs:8]: Not needed, thats in ElementID.LocalID

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs [hunk 3]
```diff
@@ -76,39 +90,39 @@ public static void MathNodeDynamicResolver(INodeReader userData, INodeWriter nod
                 {
                     // Output port gets constrained the same way.
                     var port = i == 0
-                            ? nodeWriter.AddPort<GraphType>(userData, "Out", false, registry)
-                            : nodeWriter.AddPort<GraphType>(userData, $"In{i}", true, registry);
+                        ? node.AddPort("Out", false, true)
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs:91]: Needs to be happening on concrete layer

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs (hunk 1 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs]: This is my mistake. I can change it.

**Pull request merged**